lib/rucio/core/account_counter.py

```diff
@@ -91,21 +93,24 @@ def get_updated_account_counters(total_workers, worker_number, *, session: "Sess
     :param session:            Database session in use.
     :returns:                  List of rse_ids whose rse_counters need to be updated.
     """
-    query = session.query(models.UpdatedAccountCounter.account, models.UpdatedAccountCounter.rse_id).\
-        distinct(models.UpdatedAccountCounter.account, models.UpdatedAccountCounter.rse_id)
+
+    query = (select(models.UpdatedAccountCounter.account, models.UpdatedAccountCounter.rse_id)
+             .distinct()
+             )
 
     if session.bind.dialect.name == 'oracle':
         hash_variable = 'CONCAT(account, rse_id)'''
     else:
         hash_variable = 'concat(account, rse_id)'
 
     query = filter_thread_work(session=session, query=query, total_threads=total_workers, thread_id=worker_number, hash_variable=hash_variable)
+    query_result = session.execute(query)
```

Comment:
not 100% sure here if I had to filter `query` or `query_result`

Reply:
`filter_thread_work()` takes a query and returns another one. Think of it as building on the existing query, like:

```python
query = query.where(models.UpdatedAccountCounter.rse_id == …)
```

I would like to question the introduction of `query_result` (which can be somewhat misleading because it does not contain any result. Who you oppose the following?

```python
return [row._asdict() for row in session.execute(query).all()]
```